pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -167,7 +168,6 @@ func (r *Reader) readHeader(ctx context.Context) (doneReadingFile bool) {
 	}
 	r.headerReader = nil
 	r.HeaderFinalized = true
-	r.initialBufferSize = scanner.DefaultBufferSize
```

Comment:
I think this was a bug to reset the buffer size here.

Reply:
>Header size may be larger than log line size. Why continue allocating for the header size?

What do you mean? I am confused, we use `initialBufferSize` for initiating both the header read buffer as well as the body read buffer, but we never update it, so I am not sure why here a reset is needed to the default value.

Reply:
You're right. I was thinking we update it as we read, in case we reach EOF and then later create a new scanner before reading more. This is implemented via another field though.